web/src/org/openmrs/module/chica/web/ManualCheckinSSNMRN.java

```diff
@@ -70,20 +61,18 @@ public static void verifyMRN(HttpServletRequest request, HttpServletResponse res
 			log.error("Error validating MRN: " + mrn);
 		}
 		
-		Patient patient = null;
-		Integer sessionId = null;
+		EncounterService encounterService = Context.getEncounterService();
+		List<Encounter> encounters = encounterService.getEncountersByPatientIdentifier(mrn);
+		int encounterSize = encounters.size();
+				
 		if (valid) {
 			pw.write(XML_RESULT_START + XML_TRUE + XML_RESULT_END);
-			patient = getPatientByMRN(mrn);
-			try{
-				sessionId = getEncounterSessionId(patient.getPatientId());
-			}catch(NullPointerException exp){}
 		} else {
 			pw.write(XML_RESULT_START + XML_FALSE + XML_RESULT_END);
 		}
 		
 		boolean sessionValid = false;
-		if (sessionId != null) {
+		if (encounterSize != 0) {
 			sessionValid = true;
 		}
 		

```

Comment:
This is just my own personal opinion, but I'd rather get rid of lines 66 and  74 - 76 and just do if(encounters.size() > 0)